pkg/util/kubernetes/apiserver/metadata_test.go

```diff
@@ -32,11 +32,6 @@ func TestMetadataControllerSyncEndpoints(t *testing.T) {
 
 	metaController, informerFactory := newFakeMetadataController(client)
 
-	stop := make(chan struct{})
-	defer close(stop)
-	informerFactory.Start(stop)
-	go metaController.Run(stop)
-
```

Comment:
Essentially calling `informerFactory.Start(stop)` runs over the objects manually added to the store since they don't exist in the fake apiserver.